integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanOriginationValidationTest.java

```diff
@@ -172,6 +172,80 @@ public void uc2() {
+                    .createLoanProduct(createOnePeriod30DaysLongNoInterestPeriodicAccrualProductWithAdvancedPaymentAllocation()
+                            .loanScheduleType(LoanScheduleType.PROGRESSIVE.toString()));
+            final PostLoansRequest applicationRequest01 = applyLoanRequest(client.getClientId(), loanProductResponse.getResourceId(),
+                    operationDate, 100.0, 5).numberOfRepayments(6)//
+                    .loanTermFrequency(6)//
+                    .loanTermFrequencyType(2)//
+                    .transactionProcessingStrategyCode(null)//
+                    .repaymentEvery(1)//
+                    .repaymentFrequencyType(2)//
+            ;//
+            loanTransactionHelper.applyLoanWithError(applicationRequest01, 400);
+            CallFailedRuntimeException callFailedRuntimeException = Assertions.assertThrows(CallFailedRuntimeException.class,
+                    () -> loanTransactionHelper.applyLoan(applicationRequest01));
+            // Client Id null
+            Assertions.assertTrue(
+                    callFailedRuntimeException.getMessage().contains("The parameter `transactionProcessingStrategyCode` is mandatory."));
+
+            final PostLoansRequest applicationRequest02 = applyLoanRequest(client.getClientId(), loanProductResponse.getResourceId(),
+                    operationDate, 100.0, 5).numberOfRepayments(6)//
+                    .clientId(null) //
+                    .loanTermFrequency(6)//
+                    .loanTermFrequencyType(2)//
+                    .transactionProcessingStrategyCode(LoanProductTestBuilder.ADVANCED_PAYMENT_ALLOCATION_STRATEGY)//
+                    .repaymentEvery(1)//
+                    .repaymentFrequencyType(2)//
+            ;//
+            loanTransactionHelper.applyLoanWithError(applicationRequest02, 400);
+            callFailedRuntimeException = Assertions.assertThrows(CallFailedRuntimeException.class,
+                    () -> loanTransactionHelper.applyLoan(applicationRequest02));
+            Assertions.assertTrue(callFailedRuntimeException.getMessage().contains("The parameter `clientId` is mandatory."));
+
+            // Submitted Date null
+            final PostLoansRequest applicationRequest03 = applyLoanRequest(client.getClientId(), loanProductResponse.getResourceId(),
+                    operationDate, 100.0, 5).numberOfRepayments(6)//
+                    .submittedOnDate(null) //
+                    .loanTermFrequency(6)//
+                    .loanTermFrequencyType(2)//
+                    .transactionProcessingStrategyCode(LoanProductTestBuilder.ADVANCED_PAYMENT_ALLOCATION_STRATEGY)//
+                    .repaymentEvery(1)//
+                    .repaymentFrequencyType(2)//
+            ;//
+            loanTransactionHelper.applyLoanWithError(applicationRequest03, 400);
+            callFailedRuntimeException = Assertions.assertThrows(CallFailedRuntimeException.class,
```

Comment:
Please dont do this.
Use the following to check: `assertEquals(400, callFailedRuntimeException.getResponse().code());`